src/components/List/List.tsx

```diff
@@ -120,12 +119,40 @@ class List extends UIComponent<Extendable<ListProps>, FocusContainerState> {
     )
   }
 
+  componentDidMount() {
+    this.focusHandler = new ContainerFocusHandler(
+      () => this.props.items.length,
+      index => {
+        const targetComponent = this.itemRefs[index] && this.itemRefs[index].current
+        const targetDomNode = ReactDOM.findDOMNode(targetComponent) as any
+
+        targetDomNode && targetDomNode.focus()
+      },
+    )
+  }
+
   renderItems() {
     const { items, renderItem } = this.props
-    const itemProps = _.pick(this.props, List.itemProps)
 
+    this.itemRefs = []
     return _.map(items, (item, idx) => {
-      itemProps.focusableItemProps = this.focusContainer.createItemProps(idx, items.length)
+      let maybeSelectableItemProps = {}
```

Comment:
nit, you may initialize it as

```js
const maybeSelectableItemProps = {} 
```

and then on line 145
```js
maybeSelectableItemProps.tabIndex = 0
maybeSelectableItemProps.ref = _ref
maybeSelectableItemProps.onFocus = () => this.focusHandler.syncFocusedItemIndex(idx)
```
So the no rebinding will happen for the object